### PR TITLE
Complete the restructure of binaries - Closes #192

### LINKED ISSUE #192: Rearchitect the structure of Binaries
Currently, we shove all dependencies into the same folder and ship it. Instead, we should move to the following format:

```
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:33 backup
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:33 mainnet
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:33 testnet
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:28 node
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:28 pgsql
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:28 db_data
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:30 lib
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:33 .pm2
drwxrwxr-x  2 lisk lisk 4096 Aug 23 11:36 bin
-rw-rw-r--  1 lisk lisk    0 Aug 23 11:29 lisk.sh
-rw-rw-r--  1 lisk lisk    0 Aug 23 11:29 lisky.sh
-rw-rw-r--  1 lisk lisk    0 Aug 23 11:35 bridge.sh
-rw-rw-r--  1 lisk lisk    0 Aug 23 11:29 env.sh
-rw-rw-r--  1 lisk lisk    0 Aug 23 11:33 config.json-mainnet
-rw-rw-r--  1 lisk lisk    0 Aug 23 11:33 config.json-testnet
```

- `testnet` and `mainnet` folders included for both networks
- `node` folder for nodejs binaries
- `pgsql` folder for postgres binaries
- `db_data` for database data storage independent of postgres
- `lib` folder to install “global” dependencies, such as pm2 and lisky
- `.pm2` folder for storing PM2 God daemon and logs
- `bin` folder for storing tools like JQ and other symlinks

Scripts are standard. `config.json` files are stored independent of the installation, but updated from the new copy when its deployed.

Deployment will have to change to enable this new format. The binaries will now be packaged individually, per dependency, so that they can be individually updated if needed, cutting down on the required download size for each release.

Task List:

- [x] Decouple binaries from `root-dir`
- [x] Use full paths for the build process to make following flow easier
- [x] Decouple lisk.sh from `root-dir`
- [x] Decouple database storage from pgsql directory into `blockchain.db`
- [x] Migrate binaries such as JQ to `bin`
- [x] Modify `build.sh` to follow new system
- [x] Build testnet and mainnet binaries independently
- [x] Modify `env.sh` to work with the new structure

- [x] Package testnet and mainnet binaries independently
- [x] Package postgres, node and redis binaries independently

Tasks to address in separate issues:

- Modify `installLisk.sh` to be aware of individual binary structure for updates - Issue #195 
- Migrate config.files to `root-dir` - Requires changes in core first

### PR DESCRIPTION
**Pull request by Isabello**
Resolves the last two tasks for #192 and provides pathway to install/lisk.sh rewrite.

### CODE REVIEW

File: scripts/env.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 cd "$(cd -P -- "$(dirname -- "$0")" && pwd -P)" || exit 2
 
 # Binary paths
-export PATH="$PWD/bin:$PWD/pgsql/bin:$PWD/node/bin:$PATH"
+export PATH="$PWD/bin:$PWD/pgsql/bin:$PWD/node/bin:$PWD/lisky/bin:$PWD/pm2/bin$PATH"
```
Comment (willclarktech) [inline scripts/env.sh]: Missing a colon

Comment (willclarktech) [reply scripts/env.sh]: But actually global npm installation should add the lisky executable to PATH anyway.

Comment (Isabello) [reply scripts/env.sh]: nice catch thank you :

Comment (Isabello) [reply scripts/env.sh]: No. You are misunderstanding what this does for us.

env.sh creates a fake environment setup to store all binary stuff in one place without touching the OS.

**Review by MaciejBaj — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**